### DISCUSSION TITLE
genres of top100

### DISCUSSION THREAD

**Discussion by rafalsza** [cinemagoer/cinemagoer#408] [Q&A]
Hi
How can I append a genre for all top100 movies?

`class ImdbscraperPopular:
   
    def __init__(self):
        self.df = None
        self.results = None

    def parse(self):
        ia = Cinemagoer()
        self.results = []
        top100 = ia.get_popular100_movies()

        for i in top100:

            rank = i.get('popular movies 100 rank')
            name = i.get('title')
            year = i.get('year')
            rating = i.get('rating')
            num_reviews = i.get('votes')
            genre = ?

            self.results.append({'rank': rank,
                                 'movie': name,
                                 'year': year,
                                 'rating': rating,
                                 'genre':genre,
                                 'num_reviews': num_reviews})
            self.df = pd.DataFrame(self.results)`

**Comment by alberanid** [comment]
pardon the late reply.

When you get a list of titles for a search or a top/bottom list, you only get some minimal information.

Once you have a `Movie` instance you can always populate the missing info with something like this (using the same variables as in your code):
```
ia.update(i)
genre = i.get('genre')
```

Beware the genre is a list of strings.

Hope this help.

**Reply by rafalsza** [reply]
```python
class IMDBSCRAPER:
    results = []

    def parse(self):
        ia = Cinemagoer()
        top250 = ia.get_top250_movies()
        for i in top250:
            rank = i.get('top 250 rank')
            name = i.get('title')
            year = i.get('year')
            rating = i.get('rating')
            num_reviews = i.get('votes')
            ia.update(i)
            genre = i.get('genre')

            self.results.append({'rank': rank,
                                 'movie': name,
                                 'year': year,
                                 'genre': genre,
                                 'rating': rating,
                                 'num_reviews': num_reviews})
            self.df = pd.DataFrame(self.results)
``` 

it takes a long time to generate such a list

**Reply by alberanid** [reply]
you can reduce the parsed pages to just the main one with `ia.update(i, info='main')` but fetching a single movie at a time you can't expect amazing performances.

**Comment by alberanid** [comment]
This discussion is no longer relevant due to issue https://github.com/cinemagoer/cinemagoer/issues/557: as of June 2026 Cinemagoer no longer analyzes web pages and can only rely on non-commercial datasets.
